design-documents/web-api-controllers.md

```diff
@@ -117,17 +139,25 @@ and GraphQL resolvers would be able to include such logic but it would not be re
 could be performed by the service contract responsible for posting order but service contracts are not supposed to know
 about authenticated users.
  
-The solution to these problems would be to introduce reusable area-agnostic user input processors that would perform
+The solution to these problems would be to employ reusable area-agnostic context aware classes that would perform
 such actions. These would be arbitrary classes with no common interface but they would be reusable across different areas
-and reside inside a special namespace `Presentation` to inform developers that this classes belong to presentation
+and reside inside a special namespace `ContextAware` to inform developers that this classes belong to presentation
```

Comment:
This name is confusing too. Similar classes are already located in `Controller` namespace, why don't we use existing namespace for these purposes? It also prevents potential confusion caused by putting such logic into `Model` namespace.

Reply:
Do you propose to put these classes that will be usable across different areas like REST, GraphQL, CLI into Controller namespace? A sub-namespace like Controller/Helper?

Reply:
Yes, not `Helper` however. E.g.`Controller/Validator`

Reply:
I describe multiple cases when we would create classes working with contexts like request, session, authenticated user reading HTTP request object, providing data for templates/response objects, doing authorization and validation. Those are not just validators

Reply:
furthermore Controller namespace reserved for HTML area controllers which are being picked up by name by our framework depending on URI requested. That might create some problems for us